lua/nvim-tree/actions/change-dir.lua

```diff
@@ -19,6 +20,7 @@ function M.fn(name, with_open)
 
   local foldername = name == ".." and vim.fn.fnamemodify(utils.path_remove_trailing(core.get_cwd()), ":h") or name
   local no_cwd_change = vim.fn.expand(foldername) == core.get_cwd()
+    or not M.options.global and M.options.restrict and foldername < vim.fn.getcwd(-1, -1)
```

Comment:
shouldn't it be `core.get_cwd()` instead of vim.fn.getcwd ?

Reply:
> shouldn't it be `core.get_cwd()` instead of vim.fn.getcwd ?

I think I used that so that it works with the other options: global and change_cwd. I shall experiment further.

The question is whether you are happy for this to go in; there is an alternative for the user: https://github.com/kyazdani42/nvim-tree.lua/issues/1037#issuecomment-1100576089

Reply:
We definitely want to check against the global cwd, not core's, which may be underneath.

Tested with permutations of change_cwd and global.

global works exactly as documented: the user cannot change a directory upwards. We could disable restrict_above_cwd when global is set, however I prefer to let the user configure this.